dspy/primitives/program.py

```diff
@@ -19,6 +21,12 @@ def __init__(self, callbacks=None):
 
     @with_callbacks
     def __call__(self, *args, **kwargs):
+        if settings.track_usage and settings.usage_tracker is None:
+            with track_usage() as usage_tracker:
+                output = self.forward(*args, **kwargs)
+                output.lm_usage = usage_tracker.get_total_tokens()
```

Comment:
q: what happens if the signature of the program is `xxx -> lm_usage`?

Reply:
aha good point! Although I think no user will do this, we can actually make it more robust by changing property to a method. Let me fix.